implant/sliver/transports/httpclient/drivers/win/tools/defines.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+func format(str string) string {
+	if !strings.Contains(str, "_") {
+		return str
+	}
+
+	str = strings.ReplaceAll(strings.ToLower(str), "_", " ")
+	str = strings.ReplaceAll(strings.Title(str), " ", "")
+	str = strings.ReplaceAll(str, "Crlf", "CRLF")
+	str = strings.ReplaceAll(str, "Http", "HTTP")
+
+	return str
+}
+
+func genFile(
+	pkg string,
+	cache map[string]string,
+	lines [][]string,
+) error {
+	var e error
+	var file string = "generated.go"
+	var g *os.File
+	var sorted []string
+
+	if g, e = os.Create(file); e != nil {
+		return fmt.Errorf("failed to create %s: %w", file, e)
+	}
+	defer g.Close()
+
+	g.WriteString(
+		"// Code generated by tools/defines.go; DO NOT EDIT.\n",
+	)
+	g.WriteString("package " + pkg + "\n\n")
+	g.WriteString("const (\n")
+
+	for k := range cache {
+		sorted = append(sorted, k)
+	}
+
+	sort.Slice(
+		sorted,
+		func(i int, j int) bool {
+			return len(sorted[i]) > len(sorted[j])
+		},
+	)
+
+	for _, line := range lines {
+		for _, k := range sorted {
+			line[1] = strings.ReplaceAll(line[1], k, cache[k])
+		}
+
+		if strings.HasPrefix(line[1], "\"") {
+			g.WriteString(
+				fmt.Sprintf("\t%s string = %s\n", line[0], line[1]),
+			)
+		} else if strings.HasPrefix(line[1], "L\"") {
+			g.WriteString(
+				fmt.Sprintf(
+					"\t%s string = %s\n",
+					line[0],
+					strings.Replace(line[1], "L", "", 1),
+				),
+			)
+		} else if strings.HasPrefix(line[1], "TEXT(") {
+			line[1] = strings.Replace(line[1], "TEXT(", "", 1)
+			line[1] = strings.Replace(line[1], ")", "", 1)
+
+			g.WriteString(
+				fmt.Sprintf("\t%s string = %s\n", line[0], line[1]),
+			)
+		} else {
+			g.WriteString(
+				fmt.Sprintf("\t%s uintptr = %s\n", line[0], line[1]),
+			)
+		}
+	}
+
+	g.WriteString(")\n")
+
+	return nil
+}
+
+func init() {
+	flag.Parse()
+}
+
+func main() {
+	var cache = map[string]string{
+		"NULL":   "0",
+		"sizeof": "len",
+	}
+	var lines [][]string
+
+	if flag.NArg() == 0 {
+		return
+	}
+
+	for i, arg := range flag.Args() {
+		if (i == 0) || !DoesExist(arg) {
+			continue
+		}
+
+		if e := processFile(&cache, &lines, arg); e != nil {
+			panic(e)
+		}
+	}
+
+	if e := genFile(flag.Arg(0), cache, lines); e != nil {
+		panic(e)
+	}
+}
+
+func processFile(
+	cache *map[string]string,
+	lines *[][]string,
+	file string,
+) error {
+	var b []byte
+	var e error
```

Comment:
Please use `err` for error values as is convention in Golang. You should also generally* avoid single character variable names as it makes the code harder to read.